### ISSUE TITLE
Let me enter my negative test

### ISSUE THREAD

**Issue by ndegendogo** [corona-warn-app/cwa-wishlist#239]
## Feature description

I want a dialog where I can enter my "negative" test result. At any time, not related to any prior DEK key upload. This fact will *not* be submitted to the server, it stays locally on my device, but has an impact on my risk assessment / risk status.

No special plausibility checks needed here. Can be a QR code or even manual input. The lab needs not even be integrated to cwa interface.

## Problem and motivation

Scenario:
- on day 1, I have a contact to an infected person
- the person gets tested positive on day 2, and shares their DEK keys with cwa
- I download these keys on day 3 and get a red warning
- I am a good citizen. I go home at once, self-quarantine, and get advice from my doctor / the health authorities.
- on day 4 I get a lab test, and luckily it is negative. 
- on day 5 I go back to work and out into the dangerous world with infection numbers exploding around me.

But my cwa warning is still red, and will stay so for another 9 days till my risky contact fades out. And I know that I have mitigated the initial reported risk with the lab test, but now I am blind, and will get no further warning beyond the already-red screen.
And this feels weird, especially as I have used cwa actively now for 4 1/2 months.

I want to tell cwa that I have mitigated the initial risk, and I want it to ignore this risky contact, and turn green again, and to warn me again of the next risky contact. And I want this now, without having to wait another 9 days.

Of course the details of such a feature need to be adviced by RKI. My naive assumption is that cwa could ignore the key files older than the lab sample taken, plus maybe a few days for the incubation period. Even better if it could filter the encounters by this condition (but I guess this would have to be implemented in ENF?)

**Comment by dsarkar** [comment]
Dear @ndegendogo,

thank you very much for your contribution. There are actually several discussion already on the topic of negative test results, both in similar and also different contexts, see below. Your specific issue has been raised here https://github.com/corona-warn-app/cwa-wishlist/issues/115 and an internal tracking ticket exists. I would like to ask you to continue any further discussion on this issue there.

General information:
https://www.coronawarn.app/de/faq/#test_negative 
https://www.coronawarn.app/en/faq/#test_negative 
https://github.com/corona-warn-app/cwa-faq/issues/315  

Discussions regarding negative test results:
https://github.com/corona-warn-app/cwa-wishlist/issues/115 
https://github.com/corona-warn-app/cwa-documentation/issues/326 
https://github.com/corona-warn-app/cwa-documentation/issues/377 
https://github.com/corona-warn-app/cwa-documentation/issues/381 
https://github.com/corona-warn-app/cwa-documentation/issues/65 

Best regards,
DS

Corona-Warn-App Open Source Team

**Comment by dsarkar** [comment]
Duplicate of https://github.com/corona-warn-app/cwa-wishlist/issues/115

**Comment by daimpi** [comment]
@dsarkar I don't think this is a duplicate. #115 deals with resetting your status after you tested **positive** and went through quarantine (i.e. having a red status triggered by a positive test), while this request is about getting a red exposure warning in CWA but testing **negative**.

Somewhat related: [Request for better information to distinguish red and green encounters](https://github.com/corona-warn-app/cwa-wishlist/issues/233).

**Comment by dsarkar** [comment]
Dear @ndegendogo, dear @daimpi,

apologies,  I re-open the issue, since it is not an exact duplicate.

Best regards,
DS

Corona-Warn-App Open Source Team

**Comment by ndegendogo** [comment]
Thanks @dsarkar

**Comment by dsarkar** [comment]
Dear @ndegendogo, have a look also here, please: https://github.com/corona-warn-app/cwa-app-ios/issues/1427#issuecomment-718695647.

**Comment by nschloe** [comment]
From https://github.com/corona-warn-app/cwa-app-ios/issues/1427#issuecomment-718733101:

> The current situation has another significant disadvantage. If the card stays red for 14 days no matter what, the user will not know if she had another high-risk encounter after a negative test. If, on the other hand, the card turns green after the test and red again a few days later, it's clear that attention is required once again.

**Comment by daimpi** [comment]
@dsarkar thanks for linking the other thread 🙂. I think it's clear to us that the current behavior is not strictly speaking a "bug" but rather an annoying "feature" of CWA. This is why this issue is in very good hands here in the wishlist repo and https://github.com/corona-warn-app/cwa-app-ios/issues/1427 can be closed as a duplicate to continue the discussion here imho.

@nschloe Yes exactly. That's why the OP states:
> […] I want to tell cwa that I have mitigated the initial risk, and I want it to ignore this risky contact, and turn green again, and to warn me again of the next risky contact.

And as mentioned above: a somewhat related [request for better information to distinguish red and green encounters](https://github.com/corona-warn-app/cwa-wishlist/issues/233).

**Comment by heinezen** [comment]
Hello everyone,

We have created a Jira ticket (EXPOSUREAPP-3496) for the feature request. It will now be investigated by the developers and the RKI. I have also expanded @ndegendogo 's issue title to be more descriptive. I hope that's OK for you.

Since https://github.com/corona-warn-app/cwa-app-ios/issues/1427 was very similar both threads have been integrated into the Jira ticket. I will close https://github.com/corona-warn-app/cwa-app-ios/issues/1427 as a duplicate of this issue, so that we can discuss the issue in one place.

Regards,
CH

----
Corona-Warn-App Open Source Team

**Comment by Ein-Tim** [comment]
@ndegendogo

>  And I know that I have mitigated the initial reported risk with the lab test, but now I am blind, and will get no further warning beyond the already-red screen.

Do you know if this is still the case? If a user got a high risk notification and gets another one then, will they get another notification or not? 

cc @thomasaugsten

**Comment by ndegendogo** [comment]
> If a user got a high risk notification and gets another one then, will they get another notification or not?

As far as I understand, the notification is currently sent on a transition green -> red or vice versa.
**And, yes, I would appreciate a lot if it could send another notification if it finds a new red contact on a later day than before.**

**Comment by Ein-Tim** [comment]
FYI: It seems like a positive test result will lead to a red risk card in the future ([PR](https://github.com/corona-warn-app/cwa-app-android/pull/4893)).
If this is implemented, it only makes sense that if a negative test is added, the risk card will turn green.

Is this planned @thomasaugsten?

**Comment by thomasaugsten** [comment]
It is not planned that test results will affect the risk card

**Comment by Ein-Tim** [comment]
@thomasaugsten What exactly is this PR doing then? https://github.com/corona-warn-app/cwa-app-android/pull/4893

**Comment by larswmh** [comment]
Closing this ticket as it was internally set to obsolete. As mentioned above:

> It is not planned that test results will affect the risk card

--- 

> What exactly is this PR doing then? https://github.com/corona-warn-app/cwa-app-android/pull/4893

@Ein-Tim I've seen you've asked this in the PR as well and for completeness, I'll post the answer here again:

> - The risk calculation is not stopped anymore if a positive test result is available
> - The risk card is only `hidden` if
>   - The risk card would show no risk or a low risk (green card) AND
>   - There is at least one positive test which is not older than n days (n = config value)
>     -The age of a PCR test is calculated from the test registration time
>     - The age of a RAT test is calculated from the test execution time
